framework/src/modules/pos/module.ts

```diff
@@ -577,7 +577,7 @@ export class PoSModule extends BaseModule {
 	private async _createStakeWeightSnapshot(context: BlockAfterExecuteContext): Promise<void> {
 		const snapshotHeight = context.header.height + 1;
```

Comment:
I think that it should be `snapshotHeight = context.header.height`

Reply:
In the last block of round `r` you set the entry for `snapshotStore(r)`. Then, at the last block of round `r+2`, you use `snapshotStore(r)` to calculate validators for round `r+3`.

See also definition of snapshot substore 

> Each store key is roundNumber.to_bytes(4,'big'), i.e., the big endian uint32 serialization of roundNumber, where roundNumber is the number of the round at the end of which the active validators and weights are computed.

Reply:
We agreed to use the notation `snapshot(r+3)` for the snapshot taken at the last block of round `r` therefore the comments above are outdated; current version seems correct and the comment can be resolved.